packages/presets/src/ai/config/edgeless-copilot.ts

```diff
@@ -308,7 +314,104 @@ const chatWithAI: AIItemConfig = {
 const makeItReal: AIItemConfig = {
   name: 'Make it real',
   icon: MakeItRealIcon,
-  showWhen: () => true,
+  showWhen: (_, editorMode, host: EditorHost) => {
+    return (
+      editorMode === 'edgeless' &&
+      showWhen(host, service => {
+        const selected = getSelectedElements(service);
+        return selected.length > 0;
+      })
+    );
+  },
+  handler: async host => {
+    const rootBlockId = host.doc.root?.id;
+    assertExists(rootBlockId);
+
+    const aiPanel = host.view.getWidget(AFFINE_AI_PANEL_WIDGET, rootBlockId);
+    const copilotPanel = host.view.getWidget(
+      AFFINE_EDGELESS_COPILOT_WIDGET,
+      rootBlockId
+    );
+    const selectedElements = getSelectedElements(getService(host));
+
+    if (!(aiPanel instanceof AffineAIPanelWidget && selectedElements.length)) {
+      return;
+    }
+
+    const edgelessRoot = getEdgelessRootFromEditor(host);
+    const { notes, frames, shapes, images } =
+      BlocksUtils.splitElements(selectedElements);
+    if (notes.length + frames.length + images.length + shapes.length === 0) {
+      return;
+    }
+    const canvas = await edgelessRoot.clipboardController.toCanvas(
+      [...notes, ...frames, ...images],
+      shapes
+    );
+    if (!canvas) {
+      return;
+    }
+
+    const png = canvas.toDataURL('image/png');
+
+    if (!png) return;
+
+    assertType<EdgelessCopilotWidget>(copilotPanel);
+    copilotPanel.hide();
+
+    const generateAnswer: AffineAIPanelWidgetConfig['generateAnswer'] = ({
+      finish,
+      input,
+      signal,
+      update,
+    }) => {
+      if (!AIProvider.actions.makeItReal) return;
+      const stream = AIProvider.actions.makeItReal({
+        input,
+        stream: true,
+        docId: host.doc.id,
+        workspaceId: host.doc.collection.id,
+        attachments: [png],
+      });
+      bindEventSource(stream, { update, finish, signal });
+    };
+    // const selectionRect = copilotPanel.selectionModelRect;
+
+    aiPanel.config = {
```

Comment:
`AffineAIPanelWidgetConfig` is initialized one time at https://github.com/toeverything/blocksuite/pull/6728/files#diff-19bc5e6df24f77683ae5a0429af2241650ac3c24af898f92e78da8b46f826a26R40

I think we need to reconsider how the config can be used in both edgeless and doc mode

Reply:
Instead of exposing config, we can provide a method to configure it.